src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala

```diff
@@ -7,14 +7,19 @@ import org.apache.mesos.Protos.TaskID
 /**
   * Utility functions for dealing with TaskIDs
   */
-
+// TODO: No one should need a util in order to fetch tasks from TaskRepo
+// there should rather be a layer that provides access to tasks by appId and taskId
 class TaskIdUtil {
-  val taskDelimiter = "."
+  val appDelimiter = "."
   val TaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
   val uuidGenerator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())
 
+  // TODO: using a colon here breaks EntityRepository.allIds() because that only collects names w/o ":"
+  // changing this does not really remove the problem and would require a migration
+  val idDelimiter = ":"
```

Comment:
Indeed, I got rid of a lot of binary functions. Most of them now only take the taskId, which makes mapping easier in various occasions.